### PR TITLE
Multiple Dropper v0.2.0 API improvements

### PR DESCRIPTION
**Pull request by zomglings**
### Queries and responses on `/contracts/requests` now show `contract_address`

Now the caller can specify `contract_id` or `contract_address` as a query parameter.

At least one of those parameters should be provided.

`registered_contract_id` in response is now `contract_id`.

The response also contains a `contract_address` field.

Resolves https://github.com/bugout-dev/engine/issues/283

### Create `call_requests` using contract ID or contract address

This change allows users to create call requests by posting to `/contracts/requests` with a JSON body which contains a `"contract_id"` key or a `"contract_address"` key.

This change also removes the old `/contracts/<contract_id>/requests` POST endpoint.

Resolves https://github.com/bugout-dev/engine/issues/298

### Moved `POST /contracts/register` to `POST /contracts/`

This makes the API more RESTful.

### `POST /contracts/` documentation shows `contract_type` as an enum

Previously, this parameter of the request body was being displayed as a string.

Resolves https://github.com/bugout-dev/engine/issues/276

### Allow users to edit registered contracts

added a PUT `/contracts/<contract_id>` endpoint, which allows users to modify the title, description, and image URI for a contract. These changes only apply to the Moonstream Engine API.

Resolves https://github.com/bugout-dev/engine/issues/296

### Updated `waggle` to respect new API changes

- [x] Queries and responses on `/contracts/requests`
- [x] Create `call_requests` using contract ID or contract address

### Unique constraint on `registered_contracts` modified to include `moonstream_user_id`

This allows different users to register the same contract under their respective accounts.

Resolves https://github.com/bugout-dev/engine/issues/295

### `ttl_days` is not longer directly inserted into raw SQL

Also fixed a bug on how the list call requests endpoint handled call requests with `null` `expires_at` fields - it led to a 500 error before.

Resolves https://github.com/bugout-dev/engine/issues/300

## How to test these changes?

Tested manually.

### CODE REVIEW

**Comment by zomglings** [comment]
@bugout-dev check

- [ ] Run migration on production database

File: api/engineapi/routes/contracts.py [hunk 1]
```diff
@@ -133,6 +130,32 @@ async def register_contract(
...
+        contract = contracts_actions.update_registered_contract(
+            db_session,
+            request.state.user.id,
+            contract_id,
+            update_info.title,
+            update_info.description,
+            update_info.image_uri,
+            update_info.ignore_nulls,
+        )
+    except NoResultFound:
```
Comment (kompotkot) [inline api/engineapi/routes/contracts.py:150]: I suggest to handle common exceptions as in code

```python
    except Exception as err:
        logger.error(repr(err))
        raise EngineHTTPException(status_code=500)
```

File: api/engineapi/models.py [hunk 2]
```diff
@@ -160,6 +160,7 @@ class RegisteredContract(Base):  # type: ignore
     __table_args__ = (
         UniqueConstraint(
             "blockchain",
+            "moonstream_user_id",
```
Comment (kompotkot) [inline api/engineapi/models.py:163]: Are we sure if we run migration we will not get error bacause of hitting uniqe constraint?

Comment (zomglings) [reply api/engineapi/models.py:163]: Yeah, this feature was just launched into production and there are only two rows in production database under this relation currently.

**Review by kompotkot — APPROVED** [review]
lg

**Comment by zomglings** [comment]
Migration complete: https://bugout.dev/app/personal/9faa9429-aaa4-4d0b-a68d-db9fbb65ad71/entries/388dd799-d1a3-4a4e-836a-f54fff36bd2d/

**Pull request merged**